common/src/main/java/dev/isxander/yacl3/gui/utils/ItemRegistryHelper.java

```diff
@@ -0,0 +1,103 @@
+package dev.isxander.yacl3.gui.utils;
+
+
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.Items;
+
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+public class ItemRegistryHelper {
+
+	private static Item getItemFromNameImpl(String value) {
+		String namespace, itemName;
+		int sep = value.indexOf(ResourceLocation.NAMESPACE_SEPARATOR);
+		value = value.toLowerCase();
+		if (sep == -1) {
+			namespace = ResourceLocation.DEFAULT_NAMESPACE;
+			itemName = value;
+		} else {
+			namespace = value.substring(0, sep);
+			itemName = value.substring(sep + 1);
+		}
+		ResourceLocation identifier = new ResourceLocation(namespace, itemName);
```

Comment:
`ResourceLocation` already does this with `ResourceLocation(String)` or `ResourceLocation.tryParse(String)`. It decomposes it for you.